src/components/App/App.tsx

```diff
@@ -0,0 +1,194 @@
+/* eslint-disable no-console */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { AuthContext } from '../Auth/AuthContext';
+import { Todo } from '../../types/Todo';
+import * as Api from '../../api/todos';
+import { TodoList } from '../TodoList';
+import { FilterBy } from '../../types/filterBy';
+import { Footer } from '../Footer';
+
+export const App: React.FC = () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const user = useContext(AuthContext);
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoList, setTodoList] = useState<Todo[] | null>(null);
+  const [showFooter, setShowFooter] = useState(false);
+  const [error, setError] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [loadingInput, setLoadingInput] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const getTodosFromServer = useCallback(
+    () => {
+      if (user) {
+        Api.getTodos(user.id)
+          .then((data) => {
+            setTodoList(data);
+
+            if (data.length) {
+              setShowFooter(true);
+            }
+
+            setTempTodo(null);
+          })
+          .catch(() => {
+            setError('Error 404 unable to get todos');
+            setTimeout(() => setError(''), 3000);
+          });
+      }
+    },
+    [],
+  );
+
+  const addTodoToServer = () => {
+    if (user && todoTitle.length) {
+      // we disable the input while trying to post the todo.
+      setLoadingInput(true);
+
+      const newTodo = {
+        userId: user.id,
+        title: todoTitle,
+        completed: false,
+      };
+
+      setTodoTitle('');
+
+      Api.addTodo(newTodo)
+        .then(response => {
+          setTempTodo({
```

Comment:
When you add the temp todo, dont give it the title. It is just a placeholder, so better not show the result until we have a response from server

```ts
setTempTodo({
  id: 0,
  userId: user.id,
  title: '',
  completed: false,
});